### PR TITLE
Adjust LLVMSupport for upstream changes

### PR DESCRIPTION
**Pull request by smeenai**

### CODE REVIEW

**Comment by smeenai** [comment]
CC @slavapestov

File: stdlib/public/runtime/LLVMSupport.cpp [hunk 1]
```diff
@@ -68,5 +75,45 @@ llvm::SmallVectorBase::grow_pod(void *FirstEl, size_t MinCapacity,
...
+#ifdef __cpp_aligned_new
+                        ,
+                        std::align_val_t(Alignment)
+#endif
+  );
+}
+
+void llvm::deallocate_buffer(void *Ptr, size_t Size, size_t Alignment) {
+  ::operator delete(Ptr
+#ifdef __cpp_sized_deallocation
```
Comment (CodaFi) [inline stdlib/public/runtime/LLVMSupport.cpp:107]: Let's spell the full call out across this ifdef.

File: stdlib/public/runtime/LLVMSupport.cpp [hunk 2]
```diff
@@ -68,5 +75,45 @@ llvm::SmallVectorBase::grow_pod(void *FirstEl, size_t MinCapacity,
...
+void SmallVectorBase<uint64_t>::grow_pod(void *FirstEl, size_t MinCapacity,
+                                         size_t TSize);
+
+// The same for allocate_buffer and deallocate_buffer, which are used by
+// DenseMap and unique_function. This is a hack. This implementation is copied
+// from llvm/lib/Support/MemAlloc.cpp and has to stay in sync with it.
+LLVM_ATTRIBUTE_RETURNS_NONNULL LLVM_ATTRIBUTE_RETURNS_NOALIAS void *
+llvm::allocate_buffer(size_t Size, size_t Alignment) {
+  return ::operator new(Size
+#ifdef __cpp_aligned_new
```
Comment (CodaFi) [inline stdlib/public/runtime/LLVMSupport.cpp:98]: Same here

**Comment by CodaFi** [comment]
Let me temporarily retract the reviews above. I have qualms about the way the LLVM patches were handled, but if we're just trying to keep in sync then addressing them would work against that goal.

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by atrick** [comment]
@smeenai Sorry I didn't see this PR before I merged in another one https://github.com/apple/swift/pull/31455. I just reverted mine so you can go ahead with this one

**Comment by atrick** [comment]
@swift-ci smoke test

**Comment by smeenai** [comment]
Thanks! Unfortunately master-next doesn't appear to be a in a great state – the smoke test failures look unrelated. I'll rebase this and try another smoke test.

**Comment by smeenai** [comment]
@atrick actually can you merge #31455 so I can merge this?

**Pull request merged**